src/ngAnimate/animateCss.js

```diff
@@ -0,0 +1,670 @@
+  var maxValue = 0;
+  var values = isString(str) ?
+    str.split(/\s*,\s*/) :
+    [];
+  forEach(values, function(value) {
+    maxValue = Math.max(parseFloat(value) || 0, maxValue);
+  });
+  return maxValue;
+}
+
+function getCssTransitionDurationStyle(duration, applyOnlyDuration) {
+  var style = TRANSITION_PROP;
+  var value = duration + 's';
+  if (applyOnlyDuration) {
+    style += DURATION_KEY;
+  } else {
+    value += ' linear all';
+  }
+  return [style, value];
+}
+
+function getCssKeyframeDurationStyle(duration) {
+  return [ANIMATION_PROP + DURATION_KEY, duration + 's'];
+}
+
+function getCssDelayStyle(delay, isKeyframeAnimation) {
+  var prop = (isKeyframeAnimation ? ANIMATION_PROP : TRANSITION_PROP) + DELAY_KEY;
+  return [prop, delay + 's'];
+}
+
+function blockTransitions(node, bool) {
+  var value = bool ? 'none' : '';
+  var key = TRANSITION_PROP + PROPERTY_KEY;
+  applyInlineStyle(node, key, value);
+  return [key, value];
+}
+
+function blockAnimations(node, bool) {
+  var value = bool ? 'paused' : '';
+  var key = ANIMATION_PROP + ANIMATION_PLAYSTATE_KEY;
+  applyInlineStyle(node, key, value);
+  return [key, value];
+}
+
+function applyInlineStyle(node, key, value) {
+  if (arguments.length == 2) {
+    value = key[1];
+    key = key[0];
+  }
+  node.style[key] = value;
+}
+
+function LocalCacheLookup() {
+  var cache = {};
+  return {
+    flush: function() {
+      cache = {};
+    },
+
+    count: function(key) {
+      var entry = cache[key];
+      return entry ? entry.total : 0;
+    },
+
+    get: function(key) {
+      var entry = cache[key];
+      return entry && entry.value;
+    },
+
+    put: function(key, value) {
+      if (!cache[key]) {
+        cache[key] = { total: 0 };
+      }
+      cache[key].total++;
+      cache[key].value = value;
+    }
+  };
+}
+
+var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
+  var gcsLookup = new LocalCacheLookup();
+  var gcsStaggerLookup = new LocalCacheLookup();
+
+  this.$get = ['$window', '$$jqLite', '$qRaf', '$timeout', '$$rAF',
+               '$$animateRunner', '$document', '$$animateOptions', '$sniffer',
+       function($window,   $$jqLite,   $qRaf,   $timeout,   $$rAF,
+                $$animateRunner,   $document,   $$animateOptions,   $sniffer) {
+
+    var parentCounter = 0;
+    function gcsHashFn(node, extraClasses) {
+      var KEY = "$$ngAnimateParentKey";
+      var parentNode = node.parentNode;
+      var parentID = parentNode[KEY] || (parentNode[KEY] = ++parentCounter);
+      return parentID + '-' + node.getAttribute('class') + '-' + extraClasses;
+    }
+
+    function computeCachedCSSStyles(node, className, cacheKey, properties) {
+      var timings = gcsLookup.get(cacheKey);
+
+      if (!timings) {
+        timings = computeCSSStyles($window, node, properties);
+        if (timings.animationIterationCount) {
+          timings.animationIterationCount = parseInt(timings.animationIterationCount || 1, BASE_TEN);
```

Comment:
I don't think that `timings.animationIterationCount || 1` is needed as you already check for the existence of `timings.animationIterationCount`